### PR TITLE
Upgrade to grantami-bomanalytics-openapi 3.1.0rc1

### PR DESCRIPTION
**Pull request by da1910**
Additional changes:

- Support 3.13
- Drop support for 3.9
- Switch to 3.12 for main python in CI

### CODE REVIEW

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
   merge_group:
 
 env:
-  MAIN_PYTHON_VERSION: '3.9'
+  MAIN_PYTHON_VERSION: '3.12'
```
Comment (Andy-Grigg) [inline .github/workflows/ci_cd.yml:13]: Other pyansys packages tend to use the oldest supported Python package for this, as far as I can tell. We should be consistent.

```suggestion
  MAIN_PYTHON_VERSION: '3.10'
```

File: .github/workflows/server_checks.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ on:
       python-version:
         description: >
           Python version used for installing and running ``pre-commit``.
-        default: '3.10'
+        default: '3.12'
```
Comment (Andy-Grigg) [inline .github/workflows/server_checks.yml:7]:
```suggestion
        default: '3.10'
```

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (da1910) [reply .github/workflows/ci_cd.yml:13]: Hmm, I made this change on all our packages...

Comment (da1910) [reply .github/workflows/ci_cd.yml:13]: https://github.com/ansys/grantami-recordlists/blob/main/.github/workflows/ci_cd.yml
https://github.com/ansys/grantami-jobqueue/blob/main/.github/workflows/ci_cd.yml
https://github.com/ansys/grantami-serverapi-openapi/blob/main/.github/workflows/build_and_test_library.yml
https://github.com/ansys/grantami-bomanalytics-openapi/blob/main/.github/workflows/build_and_test_library.yml

Comment (Andy-Grigg) [reply .github/workflows/ci_cd.yml:13]: I had a look at some other repos, e.g. [pymapdl](https://github.com/ansys/pymapdl/blob/ededf7320da0594b4b050ddeec630666c403069e/.github/workflows/ci.yml#L23), and there they're using 3.11. Maybe there is no convention, in which case this is fine.

**Review by Andy-Grigg — APPROVED** [review]

**Pull request merged**